import/meeting.py

```diff
@@ -1,31 +1,49 @@
 import util
+import google_sheets
 
-def last_update(conn):
-    query = "select max(hour_date) from hour_log where \
-                  hour_reason = 'meeting'"
-    with conn.cursor() as cursor:
-        cursor.execute(query)
-        return cursor.fetchone()[0]
+SHEET_TITLE = 'Monthly Co-op Meeting Attendance Tracking'
 
-def transform(log):
-    return { 'email': util.standardize_email(log['Email']),
-             'first_name': log['First Name'],
-             'last_name': log['Last Name'],
-             'timestamp': util.parse_gs_timestamp(log['Timestamp']),
-             'date': util.parse_gs_timestamp(log['Timestamp'])}
+def transform(row, row_idx):
+    return { 'email': util.standardize_email(row['Email']),
+             'first_name': row['First Name'],
+             'last_name': row['Last Name'],
+             'timestamp': util.parse_gs_timestamp(row['Timestamp']),
+             'date': util.parse_gs_timestamp(row['Timestamp']),
+             'database': row.get(google_sheets.DATABASE_COL),
+             'row_idx': row_idx } # Store the row's index as well, used for updating the GSheet
 
-mapping = util.read_mapping()
+def import_meeting(conn, dry_run):
+    # We use the second page of the meetings sheet.
+    sheets = google_sheets.fetch_sheets(SHEET_TITLE, 1)
+    for s in sheets:
+        result = import_sheet(conn, s, dry_run)
+    return result
 
-def import_meeting(conn, meeting_attendance):
-    last = last_update(conn)
-    owners = util.existing(conn, 'owner')
-    log_new = [l for l in meeting_attendance
-               if (not last) or l['timestamp'] > last]
-    log_ins, log_not_ins = \
-                           util.data_email_exists(mapping, log_new, owners)
+def insert_meeting(conn, row):
     # meetings are 2 hours
     query = """insert into hour_log(email, amount, hour_date, hour_reason) \
                values (%(email)s, 2, %(date)s, 'meeting')"""
     with conn.cursor() as cursor:
-        cursor.executemany(query, log_ins)
+        cursor.execute(query, row)
+
+def import_sheet(conn, sheet, dry_run):
+    mapping = util.read_mapping()
+    header_map = google_sheets.get_header_map(sheet)
+    now_str = util.get_now_str()
+    rows = google_sheets.get_all_records(sheet)
+    transformed_rows = [transform(row, row_idx) for row_idx, row in enumerate(rows)]
+    owners = util.existing(conn, 'owner')
+    log_ins, log_not_ins = \
+                       util.data_email_exists(mapping, transformed_rows, owners)
+
+    for row in log_ins:
+        # The actual row index in the GSheet will be two more than the row's position in the list.
```

Comment:
why is this? Can we add that to the comment also?

Reply:
This relates to a similar comment in the `committee` script - 

```
        # 1 indexed + account for header
        idx = row_idx + 2
```

GSheet rows are 1 indexed and our sheets also have a header row, so in the list of incoming rows, the 0 element of the list corresponds to row 2 in the sheet, etc.

I've updated the comment and also did a quick refactor, moving the +2 incrementing logic into the meeting module's `transform` method.